designs/native-ami-versioning.md

```diff
@@ -0,0 +1,121 @@
+# Native AMI Versioning
+
+Karpenter uses the latest EKS Optimized AMI releases as defaults in their AMI Requirements. With [Drift](https://karpenter.sh/preview/concepts/deprovisioning/#drift) implemented, when a new EKS Optimized AMI is released users who have enabled drift could see their fleet rolled out to the newly released AMI without warning.
+
+[EKS Optimized AMI Releases](#eks-optimized-ami-releases) are irregular and versioned differently. Without validation, irregular release cadences could introduce unwanted AMIs in production. This doc proposes two functional improvements to AMI selection logic to ease customer pain.
+
+## What could happen when a new EKS Optimized AMI is released?
+
+When a new EKS Optimized AMI is released, Karpenter could begin deprovisioning and replacing nodes. If a user wants to revert back to an older release, they need to find the corresponding AMI IDs and change their AWSNodeTemplate to pin those values by setting their AWSNodeTemplate to the following:
+
+```
+spec:
+    amiSelector:
+        aws-ids: "ami-123,ami-456"
+```
+
+Once the user wants to return to the latest release, they’ll have to restore their configurations to the previous default as the following.
+
+```
+spec:
+    amiFamily: Ubuntu # Or Bottlerocket or AL2
+```
+
+While this isn’t a difficult process, it isn't viable to require users to jump through these hoops. They must be aware of new EKS Optimized AMI releases, understand SSM aliases to fetch the AMI IDs, and modify their AWSNodeTemplate values in their CI/CD pipelines multiple times. Some users have [requested these improvements](https://github.com/aws/karpenter/issues/1495). In order to improve the experience for all users, we’ll introduce (1) Bake Time and (2) Release Version Pinning improvements.
+
+## Bake Time
+
+### 🔑 Introduce a minimumAgeDays field in the amiSelector.
```

Comment:
Days are not necessarily all the same length of time, would hours be better here and then we could use a go duration? 

I'm not sure minimumAge is a good concept for the feature we're trying to create here. ttlAfterCreated seems nicer since it matches the other ttl semantic, with the obvious downside that "Created" is somewhat ambiguous Is it SSM parameter created or AMI created? I think it needs to the SSM parameters changed since it's the standard practice to build the AMI early and then flip the SSM parameter when actually released.

Reply:
Sorry, this is from a previous version of the doc. I had changed all occurrences to `minimumAgeDuration` except this one 🤦 . Does your comment still hold after it being changed to `minimumAgeDuration`?

Reply:
Discussed offline, the SSM Parameter is updated in a varied amount of time since the image creation. Because of that we can special case this `minimumAgeDuration` to work based off the SSM Parameter creation.